src/Geometry/MappedDiscreteModels.jl

```diff
@@ -0,0 +1,104 @@
+# module ModelWithFEMaps
+
+# using Gridap
+# using Gridap.Arrays
+# using Gridap.Fields
+# using Gridap.ReferenceFEs
+# using Gridap.Geometry
+# using Gridap.CellData
+# using Gridap.FESpaces
+
+# import Gridap.ReferenceFEs.get_node_coordinates
+# import Gridap.Geometry.get_cell_node_ids
+# import Gridap.Geometry.get_reffes
+# import Gridap.Geometry.get_cell_type
+# import Gridap.Geometry.Grid
+
+# import Gridap.Geometry.get_cell_map
+# import Gridap.Geometry.get_grid
+# import Gridap.Geometry.get_grid_topology
+# import Gridap.Geometry.get_face_labeling
+
+# """
+#     MappedGrid
+
+# Reprepresent a grid with a geometrical map that is the composition of
+# a reference to a physical space map (standard)
+# and a (vector-valued) map from physical space to physical space. E.g.,
+# it can be used to include a high order map implemented by any map that is
+# a `CellField`.
+# """
+struct MappedGrid{Dc,Dp,T,M,L} <: Grid{Dc,Dp}
+  grid::Grid{Dc,Dp}
+  geo_map::T  # Composition of old map and new one
+  phys_map::M # New map in the physical space
+  node_coords::L
+  function MappedGrid(grid::Grid{Dc,Dp},phys_map) where {Dc,Dp}
+
+    @assert length(phys_map) == num_cells(grid)
+    @assert eltype(phys_map) <: Field
+
+    function _cell_vector_to_dof_vector!(dof_vector,cell_node_ids, cell_vector)
```

Comment:
`_cell_vector_to_dof_vector!` is also defined here:

https://github.com/gridap/Gridap.jl/pull/801/files#diff-e0f080c54ca96dd8908b78eaf4295f6cde7383577d30eb034966ff0e13c2c36aR23

can we avoid code replication?